File: cs/ex4_2_1.cs
class Person { public int age; }
class Student : Person {}

class Example {
    static T elder<T>(T p, T q) where T : Person {
        return (p.age >= q.age) ? p : q;
    }

    public static void Main() {
        Person p = elder<Person>(new Person(), new Person());
        Student s = elder<Student>(new Student(), new Student());
    }
}

class Group<T> where T : Person {
    T p;
    void sortByAge() {
        int age = this.p.age;
    }
}

File: java/ex4_2_1.java
class Person { int age; }
class Student extends Person {}

class A {
    static <T extends Person> T elder(T p, T q) {
        return (p.age >= q.age) ? p : q;
    }
}

class Example {
    public static void main(String[] args) {
        Person p = A.<Person>elder(new Person(), new Person());
        Student s = A.<Student>elder(new Student(), new Student());
    }
}

class Group<T extends Person> {
    T p;
    void sortByAge() {
        int age = this.p.age;
    }
}
